src/DssSpell.sol

```diff
@@ -58,74 +60,145 @@ contract DssSpellAction is DssAction {
     //
     // uint256 internal constant X_PCT_RATE = ;
 
-    // ---------- Math ----------
-    uint256 internal constant BILLION = 10 ** 9;
-
-    // ---------- Addesses ----------
-    address internal immutable MCD_PAUSE = DssExecLib.getChangelogAddress("MCD_PAUSE");
-    address internal immutable LINE_MOM  = DssExecLib.getChangelogAddress("LINE_MOM");
-
+    // ---------- Contract addresses ----------
+    GemAbstract internal immutable MKR = GemAbstract(DssExecLib.mkr());
+
+    InputConduitJarLike internal immutable MCD_PSM_PAX_A_INPUT_CONDUIT_JAR = InputConduitJarLike(DssExecLib.getChangelogAddress("MCD_PSM_PAX_A_INPUT_CONDUIT_JAR"));
+    JarLike internal immutable MCD_PSM_PAX_A_JAR                           = JarLike(DssExecLib.getChangelogAddress("MCD_PSM_PAX_A_JAR"));
```

Comment:
Except for MKR above, all other contracts are usually declared with type `address` and casting is done inline with the calls below.

Reply:
There are at least two occasions in the archive of this exact pattern, so I would say it's a mere preference
- https://github.com/makerdao/spells-mainnet/blob/c198914645a86d51cee32d35749e8561caa14996/archive/2024-02-07-DssSpell/DssSpell.sol#L78
- https://github.com/makerdao/spells-mainnet/blob/c198914645a86d51cee32d35749e8561caa14996/archive/2024-02-22-DssSpell/DssSpell.sol#L75

Reply:
If you notice, those are quite recent spells. We have calls like that in every single spell, and the established pattern is the one suggested.
Also, there's an item in the checklist that mandates it to be done like that:
https://github.com/makerdao/pe-checklists/blob/45b65c3829e66713e84af2ce89c2209df6e5c4be/spell/spell-reviewer-mainnet-checklist.md?plain=1#L309

Reply:
I'm fine with this change, however it should not be done without proper discussion and the subsequent update of the checklist.

Reply:
I didn't see this discussion earlier & left comment below
https://github.com/makerdao/spells-mainnet/pull/405#discussion_r1568868217

tldr; I don't have strong opinion on this topic but I am not sure what does [except where the archive pattern differs from this pattern](https://github.com/makerdao/pe-checklists/blob/45b65c3829e66713e84af2ce89c2209df6e5c4be/spell/spell-reviewer-mainnet-checklist.md?plain=1#L311) would mean in this case.

Reply:
It was a reply to "You can check the archive for reference". The checklist item that you linked contains `UNLESS archive patterns permit otherwise` which is what I brought up above. Don't have a strong opinion on the topic, also fine enforcing it as `address` from now on, but then we need to remove or further specify UNLESS condition in the checklist to improve this rule

Reply:
I changed everthihng to address apart from `MKR` :)